### PR TITLE
Revised `Application`.

### PR DESCRIPTION
**Pull request by m1cm1c**
Note that I changed "will" to "can" in `2.2 Target Group` because this has been criticised in the previous meeting. If you can think anything better, write a comment.

### CODE REVIEW

**Comment by jGleitz** [comment]
If you’re on it “re-enginer” was criticised (correctly, imho), can you fix that, too?

The rest lgtm.

File: doc/Requirements Specification/Parts/Application.lyx [hunk 1]
```diff
@@ -272,7 +272,7 @@ gls{PCM}
 
 \end_inset
 
-), which can hence be fixed in time.
+), which can then be fixed early on.
```
Comment (flyfl) [inline doc/Requirements Specification/Parts/Application.lyx]: I think that this part of the sentence can be deleted, so the sentence would be:
>This leads to earlier detection of arising problems (like implementation errors or
unrealistic modelling in the PCM).

I find it redundant to say that the problems can be fixed (as it is obvious) in time and the usage of "early" two times seems to be inelegent.

**Comment by m1cm1c** [comment]
You mean that the dash should be removed (I assume this since I can't find it written with only one 'e' in `Application`)?

If you enter both versions into Google, it shows you the definition of "re-engineer" in both cases. The Oxford English Dictionary [has a definition for "re-engineer"](http://www.oxforddictionaries.com/definition/english/re-engineer) and [redirects you to this definition if you enter "reengineer"](http://www.oxforddictionaries.com/definition/english/re-engineer?q=reengineer). Furthermore, according to the spell checking in FF as well as according to the one in LibreOffice, "reengineer" is wrong.

**Comment by flyfl** [comment]
The problem was the distinction between "re-engineering" and "reversed engineering".
We do not "re-engineer" a PCM, we create a PCM with the reversed engineering method.

File: doc/Requirements Specification/Parts/Application.lyx (hunk 1 above)
Comment (m1cm1c) [inline doc/Requirements Specification/Parts/Application.lyx]: How about changing it to "This leads to fast detection of arising problems (like implementation errors or unrealistic modelling in the PCM), which can then be fixed early on."? "fast" work well with "arising" and we'd avoid using "early" twice.

Comment (jGleitz) [inline doc/Requirements Specification/Parts/Application.lyx]: +1 for 

> This leads to fast detection of arising problems (like implementation errors or unrealistic modelling in the PCM), which can then be fixed early on.

**Comment by jGleitz** [comment]
as @flyfl said, the hyphen is not the issue here. It’s whether “to re-engineer” is the correct term, which both Michael and me doubt.

File: doc/Requirements Specification/Parts/Application.lyx (hunk 1 above)
Comment (flyfl) [inline doc/Requirements Specification/Parts/Application.lyx]: That's better

**Comment by RomanLangrehr** [comment]
> as @flyfl said, the hyphen is not the issue here. It’s whether “to re-engineer” is the correct term, which both Michael and me doubt.

If so, what is the correct term?

**Comment by jGleitz** [comment]
> If so, what is the correct term?

Don’t know…

**Comment by flyfl** [comment]
I would change the usage of "re-engineering" to "reversed engineering". But therefore the whole sentence should be changed.

**Comment by RomanLangrehr** [comment]
> I would change the usage of "re-engineering" to "reversed engineering". But therefore the whole sentence should be changed.

What? Re-engineering and revers enginering are two completly different things!

**Comment by jGleitz** [comment]
> What? Re-engineering and revers[e] engine[e]ring are two completly different things!

Definitely. And reverse engineering is definitely more off.

**Comment by flyfl** [comment]
That was the actual problem. In the sentence I meant:
>After developing the software with Palladio and implementing it, a static code
analysis tool like SoMoX and Beagle can be re-engineer a PCM which can
then be compared to the initial one.

The PCM is not yet created. It has to be "reversed engineered". Or did I misunderstand something?
The approach is to compare a PCM that was manually created to one that has been reversed engineered?

**Comment by jGleitz** [comment]
Okay. I think

“After developing the software with Palladio and implementing it, a static code
analysis tool like SoMoX can reverse-engineer a PCM instance which Beagle will add resource demand information to. This PCM instance can
then be compared to the initial one.”

is correct in this case.

But we still have to fix the other uses of “to re-engineer”

**Comment by acBerger** [comment]
We have to do #156 before.

**Comment by m1cm1c** [comment]
> We have to do #156 before.

@flyfl hasn't started working on #156 yet. We can merge this PR so he can work on it after my changes made it into master. If we don't get this PR merged until he starts working on it, he still can work on it using my fork.

**Comment by RomanLangrehr** [comment]
Please resolve conflicts.

**Comment by m1cm1c** [comment]
Resolved conflicts.

File: doc/Requirements Specification/Parts/Application.lyx [hunk 2]
```diff
@@ -754,7 +752,7 @@ Glspl{software architect}
 
 \end_inset
 
- will use 
+ can use
```
Comment (acBerger) [inline doc/Requirements Specification/Parts/Application.lyx]: missing a space here

File: doc/Requirements Specification/Parts/Application.lyx [hunk 3]
```diff
@@ -805,7 +802,7 @@ Glspl{system deployer}
...
```
Comment (acBerger) [inline doc/Requirements Specification/Parts/Application.lyx]: missing a space here

File: doc/Requirements Specification/Parts/Application.lyx [hunk 4]
```diff
@@ -849,7 +845,7 @@ Glspl{component developer}
...
```
Comment (acBerger) [inline doc/Requirements Specification/Parts/Application.lyx]: missing a space here

**Comment by jGleitz** [comment]
Please adopt the changes we discussed above.

File: doc/Requirements Specification/Parts/Application.lyx (hunk 3 above)
Comment (m1cm1c) [inline doc/Requirements Specification/Parts/Application.lyx]: Thank you. Fixed it.

**Comment by m1cm1c** [comment]
The only other occurrence in `Application` seems to be correct.

**Comment by jGleitz** [comment]
> The only other occurrence in Application seems to be correct.

What do you mean by that?

**Comment by m1cm1c** [comment]
>> The only other occurrence in Application seems to be correct.

> What do you mean by that?


"Beagle can be used to re-engineer source code. "

**Comment by jGleitz** [comment]
Algright. What happened to 

> This leads to fast detection of arising problems (like implementation errors or unrealistic modelling in the PCM), which can then be fixed early on.

?

**Comment by m1cm1c** [comment]
Nothing. It's still in there. Why wouldn't it?

**Comment by jGleitz** [comment]
Oh, sorry, I got confused.

lgtm

**Pull request merged**